packages/core/src/app/common/error/ErrorModal.test.tsx

```diff
@@ -0,0 +1,124 @@
+            <LocaleContext.Provider value={localeContext}>
+                <ErrorModal />
+            </LocaleContext.Provider>
+        );
+        // eslint-disable-next-line testing-library/no-node-access
+        expect(document.querySelector(".modal--error")).not.toBeInTheDocument();
+    });
+
+    it('renders request ID if available', () => {
+        error = {
+            type: 'request',
+            headers: { 'x-request-id': 'foobar' },
+        } as unknown as RequestError;
+
+        render(
+            <LocaleContext.Provider value={localeContext}>
+                <ErrorModal error={error} />
+            </LocaleContext.Provider>
+        );
+
+        expect(screen.getByText('foobar')).toBeInTheDocument();
+    });
+
+    it('overrides error message', () => {
+        render(
+            <LocaleContext.Provider value={localeContext}>
+                <ErrorModal error={error} message="error message" />
+            </LocaleContext.Provider>
+        );
+
+        expect(screen.getByText('error message')).toBeInTheDocument();
+    });
+
+    it('overrides error title', () => {
+        render(
+            <LocaleContext.Provider value={localeContext}>
+                <ErrorModal error={error} title="error title" />
+            </LocaleContext.Provider>
+        );
+
+        expect(screen.getByText('error title')).toBeInTheDocument();
+    });
+
+    it('onClose is called when button is clicked', async () => {
+        const modalOnClose = jest.fn();
+
+        render(
+            <LocaleContext.Provider value={localeContext}>
+                <ErrorModal error={error} onClose={modalOnClose} />
+            </LocaleContext.Provider>
+        );
+
+        await userEvent.click(screen.getByRole('button', {
+            name: localeContext.language.translate('common.ok_action')
+        }));
+
+        expect(modalOnClose).toHaveBeenCalled();
+    });
+
+    it('does not show error code if prop if false', () => {
+        error = {
+            type: 'request',
+            headers: { 'x-request-id': 'foobar' },
+        } as unknown as RequestError;
+
+        render(
+            <LocaleContext.Provider value={localeContext}>
+                <ErrorModal error={error} shouldShowErrorCode={false} />
+            </LocaleContext.Provider>
+        );
+
+        expect(screen.queryByText('foobar')).not.toBeInTheDocument();
+    });
+
+    it('renders html error correctly', () => {
+        error = new CustomError({
+            data: {
+                shouldBeTranslatedAsHtml: true,
+                translationKey: 'payment.ratepay.errors.paymentSourceInfoCannotBeVerified',
+            }
+        });
+
+        render(
+            <LocaleContext.Provider value={localeContext}>
+                <ErrorModal error={error} shouldShowErrorCode={false} />
+            </LocaleContext.Provider>
+        );
+
+        expect(screen.getByRole('link', { name: 'Ratepay Data Privacy Statement' })).toBeInTheDocument()
```

Comment:
Since we have localeContext here, we can use translate to get this string.

Reply:
It's a bit complicated as this is a sub string of a link.